src/api_manager/context/request_context.cc

```diff
@@ -326,6 +344,30 @@ void RequestContext::FillReportRequestInfo(
   }
 }
 
+const std::string RequestContext::FindReadClientIPAddress() {
+  auto serverConfig = service_context_->config()->server_config();
+  std::string client_real_ip_header;
+
+  if (serverConfig->has_experimental() &&
+      serverConfig->experimental().client_real_ip_header().length() > 0 &&
+      serverConfig->experimental().client_real_ip_position() != 0 &&
+      request_->FindHeader(serverConfig->experimental().client_real_ip_header(),
+                           &client_real_ip_header)) {
+    // split headers
+    std::vector<std::string> secments;
+    secments = split(client_real_ip_header, kClientIPHeaderDelimeter, secments);
+
+    if (secments.size() >
+        (std::size_t)abs(
+            serverConfig->experimental().client_real_ip_position())) {
+      int index = serverConfig->experimental().client_real_ip_position();
+      return trim(secments[(index > 0) ? index - 1 : secments.size() + index]);
```

Comment:
I prefer to 
1) have position index, as regular index, from 0 to n-1.  not from 1, to n.
2) define xff_position   inside a message so you can use has_xff_fix_poisition to check if it is specified.

Reply:
has_xff_postion_for_client_ip() doesn't exist. xff_postion_for_client_ip is not defined it will be 0 by default. xff_postion_for_client_ip is not defined we need to use client_ip.

Do we need to support only backward direction?

Reply:
Applied python slice style index